cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -70,5 +74,6 @@ func (b ProxyBuilder) GetManager() configmanager.ClientConfigManager {
 // Close closes connections used by proxy.
 func (b ProxyBuilder) Close() error {
 	b.reporter.Close()
+	b.tlsCloser.Close()
 	return b.conn.Close()
```

Comment:
could use `multiclose.Wrap(a, b, c).Close()`

Reply:
done, the reporter didn't implement `io.Closer` I have added `return error` to it.